### PR TITLE
Internal PR [Improve PartialAction View]

### PR DESCRIPTION
**Pull request by vz10**
Improve PartialAction View to handle anonymous users on get() and add… @maxsocl @derdmitry 
https://trello.com/c/jfT8rG7D/256-improve-partialaction-view-to-handle-anonymous-users-on-get-and-add-test-for-this-case

### CODE REVIEW

File: mysite/ct/tests.py [hunk 1]
```diff
@@ -691,3 +691,22 @@ def test_continue_partial(self, url_name, url_post_data):
...
+
+    def test_get_on_anonymous_user(self):
+        """
+        Test for PartialAction for anonymous user.
+        """
+        response = self.client.post(
+            reverse('ct:partial_pause'),
+        )
+        continue_url = json.loads(response.content).get('partial_url')
+        self.client.logout()
```
Comment (cmltaWt0) [inline mysite/ct/tests.py]: Do we need this? User is not logged at this point.

**Comment by cmltaWt0** [comment]
@vz10 Did you run tests on this new code?
@maxsocl I've ran them one more time now and all tests were ok

File: mysite/ct/tests.py [hunk 2]
```diff
@@ -691,3 +691,22 @@ def test_continue_partial(self, url_name, url_post_data):
...
+        result = self.client.get(continue_url)
+        self.assertContains(result, 'User is not authenticated', status_code=401)
+
+
```
Comment (cmltaWt0) [inline mysite/ct/tests.py:709]: We need a test where there is no record DB (wrong token) as well as with existent token.
Something like your previous test:

```
def test_get_on_anonymous_user_wrong_token(self):
    """
    Test that checking for Anonymous is started before query to DB.
    """
    result = self.client.get(reverse('ct:partial_continue', kwargs={'token': 'wrong_token'}))
    self.assertContains(result, 'User is not authenticated', status_code=401)
```

Even more - we can check DB query using mock:

```
@patch('ct.views.get_object_or_404')
def test_get_on_anonymous_user_wrong_token(self, get_object_or_404):
    """
    Test that checking for Anonymous is started before query to DB.
    """
    result = self.client.get(reverse('ct:partial_continue', kwargs={'token': 'wrong_token'}))
    self.assertContains(result, 'User is not authenticated', status_code=401)
    self.assertEqual(get_object_or_404.call_count, 0)  # checking that there are no queries to DB using get_object_or_404
```

File: mysite/ct/tests.py [hunk 3]
```diff
@@ -691,3 +691,30 @@ def test_continue_partial(self, url_name, url_post_data):
         self.assertFalse(PartialHashTable.objects.filter(token=token).exists())
         result = self.client.get(continue_url)
         self.assertEqual(result.status_code, 404)
+
+
```
Comment (cmltaWt0) [inline mysite/ct/tests.py:696]: PEP8 - two blank lines

File: mysite/ct/tests.py [hunk 4]
```diff
@@ -691,3 +691,30 @@ def test_continue_partial(self, url_name, url_post_data):
...
+        """
+
+        response = self.client.post(
+            reverse('ct:partial_pause'),
+        )
+        continue_url = json.loads(response.content).get('partial_url')
+        result = self.client.get(continue_url)
+        self.assertContains(result, 'User is not authenticated', status_code=401)
+
+
```
Comment (cmltaWt0) [inline mysite/ct/tests.py:709]: PEP8 - two blank lines

File: mysite/ct/tests.py [hunk 5]
```diff
@@ -691,3 +691,30 @@ def test_continue_partial(self, url_name, url_post_data):
         self.assertFalse(PartialHashTable.objects.filter(token=token).exists())
         result = self.client.get(continue_url)
         self.assertEqual(result.status_code, 404)
+
+
+    def test_get_on_anonymous_user(self, *args):
```
Comment (cmltaWt0) [inline mysite/ct/tests.py]: Why we need *args here?

**Comment by cmltaWt0** [comment]
Also we need to enhance checking for user authentications in PartialAction.get method like this:

```
def get(self, request, token):
    user = request.user
    if user.is_authenticated() and not user.groups.filter(name='Temporary').exists():
        something logic
```

And accordingly we need to change some tests - we need to manually login as a regular user.

File: mysite/ct/tests.py (hunk 5 above)
Comment (vz10) [inline mysite/ct/tests.py]: Emmmm..... I was playing with different tests and decorators and forgot to delete it

**Pull request closed without merging**